emission/core/wrapper/suggestion_sys.py

```diff
@@ -330,17 +328,21 @@ def review_start_loc_nominatim(lat, lon):
 
 '''
 def zipcode_retrieval(zipcode):
-    try:
-        # Use this API key first. 
-        url = ZIP_HOST_URL + ZIPCODE_API_KEY + ZIP_FORMAT + zipcode + ZIP_DEGREE
-        response = requests.request('GET', url=url)
-        return response.json()
+    
+    # Use this API key first. 
+    url = ZIP_HOST_URL + ZIPCODE_API_KEY + ZIP_FORMAT + zipcode + ZIP_DEGREE
+    response = requests.request('GET', url=url)
+    results = response.json()
 
-    except:
+    if "error_code" in results:
```

Comment:
so how did this ever work? or did you never get a chance to test the error condition?

Reply:
I checked it on the ipython notebook for Sam and Bill's trips and it works, since before it was going into the automatic error condition.

Reply:
I am not asking how you tested it, although that is good to know and I wish you had included it in the original PR description.

What I want to know is why you used a try/catch block in the first place.

Reply:
I didn't realize that the error would be written in the json results and thought that it would error while retrieving the results. The code still ran because the json went through since the json itself contained something like {"error_code": 429, "error_message": blah blah}.

Reply:
ok so you did not test the error condition prior to this change - you wrote the code based on assumptions of how it would work.

Reply:
This is why unit testing is so important!